cli/src/config.rs

```diff
@@ -16,56 +16,122 @@
-pub struct CommonConfigTemplate {
-    /// Consensus configuration.
-    pub consensus: ConsensusConfig,
-    /// General configuration.
-    pub general: GeneralConfig,
-}
+use std::{
+    net::SocketAddr,
+    path::{Path, PathBuf},
+};
 
 /// Part of the template configuration.
-#[derive(PartialEq, Clone, Debug, Serialize, Deserialize, Default)]
+#[derive(PartialEq, Clone, Debug, Serialize, Deserialize)]
 pub struct GeneralConfig {
     /// Count of the validator nodes in the network.
     pub validators_count: u32,
+    /// Supervisor service mode.
+    pub supervisor_mode: SupervisorMode,
 }
 
-/// Node public configurations.
-#[derive(Debug, Clone, Serialize, Deserialize)]
+/// Public configuration of the node. Is shared among validators.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub struct NodePublicConfig {
-    /// Network address.
-    pub address: String,
-    /// Public keys of a validator.
-    pub validator_keys: ValidatorKeys,
-}
-
-/// `SharedConfig` contain all public information that should be shared in the handshake process.
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct SharedConfig {
-    /// Template for common configuration
-    pub common: CommonConfigTemplate,
-    /// Public node
-    pub node: NodePublicConfig,
+    /// Consensus configuration.
+    pub consensus: ConsensusConfig,
+    /// General configuration.
+    pub general: GeneralConfig,
+    /// Public keys of the node.
+    ///
+    /// `None` when not yet generated. The keys are generated at
+    /// `generate-config` configuration step. The keys are required for the
+    /// `finalize` step.
+    pub validator_keys: Option<ValidatorKeys>,
 }
 
-/// `NodePrivateConfig` collects all public and secret keys.
-#[derive(Debug, Serialize, Deserialize)]
+/// Private configuration of the node.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub struct NodePrivateConfig {
-    /// Listen address.
+    /// Network listening address.
     pub listen_address: SocketAddr,
-    /// External address.
+    /// Remote Network address used by this node.
     pub external_address: String,
     /// Path to the master key file.
     pub master_key_path: PathBuf,
+    /// API configuration.
+    pub api: NodeApiConfig,
+    /// Network configuration.
+    pub network: NetworkConfiguration,
+    /// Memory pool configuration.
+    pub mempool: MemoryPoolConfig,
+    /// Optional database configuration.
+    #[serde(default)]
+    pub database: DbOptions,
+    /// Amount of threads used for transactions verification.
+    pub thread_pool_size: Option<u8>,
+    /// Information about peers within network.
+    pub connect_list: ConnectListConfig,
     /// Validator keys.
     #[serde(skip)]
     pub keys: Keys,
 }
+
+/// Configuration for the `Node`.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct NodeConfig {
+    /// Private configuration of the node.
+    pub private_config: NodePrivateConfig,
```

Comment:
If I understand correctly, private / public config correspond to what is called local / global config in the [Exonum docs](https://exonum.com/doc/version/latest/architecture/configuration/). If so, we should use single set of terms everywhere.